pkg/forwarder/forwarder.go

```diff
@@ -198,7 +198,7 @@ type DefaultForwarder struct {
 	domainForwarders map[string]*domainForwarder
 	domainResolvers  map[string]resolver.DomainResolver
 	healthChecker    *forwarderHealth
-	internalState    uint32
```

Comment:
It looks like all of the accesses to this variable are under `f.m.Lock()` -- why is atomic required?

Reply:
Not sure, but there was already some atomic calls on it previous to my change (the lock may have appeared after), in my PR I've just cleaned up a bit by having all uses of this variable to be atomic.
I'll give it a second look though.